pkg/kosmosctl/image/pull.go

```diff
@@ -0,0 +1,295 @@
+     kosmosctl image pull
+
+     # Pull and save images with custom config, e.g:
+     kosmosctl image pull --kosmos-version=[kosmos-image-version] --coredns-version=[coredns-image-version] --eps-version=[eps-image-version] --output=[output-dir] --container-runtime=[container-runtime]
+`))
+
+type CommandPullOptions struct {
+	Output              string
+	ImageList           string
+	ContainerRuntime    string
+	KosmosImageVersion  string
+	CorednsImageVersion string
+	EpsImageVersion     string
+	ContainerdNamespace string
+	Context             context.Context
+	DockerClient        *docker.Client
+	ContainerdClient    *containerd.Client
+}
+
+func NewCmdPull() *cobra.Command {
+	o := &CommandPullOptions{}
+	cmd := &cobra.Command{
+		Use:                   "pull",
+		Short:                 i18n.T("pull a kosmos offline installation package. "),
+		Long:                  "",
+		Example:               PullExample,
+		SilenceUsage:          true,
+		DisableFlagsInUseLine: true,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctlutil.CheckErr(o.Complete())
+			ctlutil.CheckErr(o.Validate())
+			ctlutil.CheckErr(o.Run())
+			return nil
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.StringVarP(&o.ImageList, "image-list", "l", "", "Image list of kosmos. ")
+	flags.StringVarP(&o.Output, "output", "o", "", "Path to a output path, default path is current dir")
+	flags.StringVarP(&o.ContainerRuntime, "container-runtime", "c", utils.DefaultContainerRuntime, "Type of container runtime(docker or containerd), docker is used by default .")
+	flags.StringVarP(&o.ContainerdNamespace, "containerd-namespace", "n", utils.DefaultContainerdNamespace, "Namespace of containerd. ")
+	flags.StringVarP(&o.KosmosImageVersion, "kosmos-version", "", utils.DefaultVersion, "Image list of kosmos. ")
+	flags.StringVarP(&o.CorednsImageVersion, "coredns-version", "", utils.DefaultVersion, "Image list of kosmos. ")
+	flags.StringVarP(&o.EpsImageVersion, "eps-version", "", utils.DefaultVersion, "Image list of kosmos. ")
+	return cmd
+}
+
+func (o *CommandPullOptions) Complete() (err error) {
+	if len(o.Output) == 0 {
+		currentPath, err := os.Getwd()
+		if err != nil {
+			return fmt.Errorf("current path can not find: %s", err)
+		}
+		o.Output = currentPath
+	}
+
+	switch o.ContainerRuntime {
+	case utils.Containerd:
+		o.ContainerdClient, err = containerd.New(utils.DefaultContainerdSockAddress)
+		if err != nil {
+			return fmt.Errorf("init containerd client failed: %s", err)
+		}
+	default:
+		o.DockerClient, err = docker.NewClientWithOpts(docker.FromEnv, docker.WithAPIVersionNegotiation())
+		if err != nil {
+			return fmt.Errorf("init docker client failed: %s", err)
+		}
+	}
+
+	o.Context = namespaces.WithNamespace(context.TODO(), o.ContainerdNamespace)
+	return nil
+}
+
+func (o *CommandPullOptions) Validate() error {
+	return nil
+}
+
+func (o *CommandPullOptions) Run() error {
+	// 1. pull image from public registry
+	klog.Info("Start pulling images ...")
```

Comment:
I think we may set the log level

Reply:
Okay, I'll set the output level of the log on the entire code